Packs/CrowdStrikeFalconX/Integrations/CrowdStrikeFalconX/CrowdStrikeFalconX.py

```diff
@@ -0,0 +1,685 @@
+
+from requests import Response
+
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+import requests
+import traceback
+import shutil
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+
+def convert_environment_id_string_to_int(environment_id: str) -> int:
+    """
+    Converting the string that describes the environment id into an int which needed for the http request
+    :param environment_id: one of the environment_id options
+    :return: environment_id represented by an int
+    """
+    environment_id_options = {
+        "300: Linux Ubuntu 16.04": 300,
+        "200: Android (static analysis)": 200,
+        "160: Windows 10": 160,
+        "110: Windows 7": 110,
+        "100: Windows 7": 100,
+        "64-bit": 64,
+        "32-bit": 32,
+    }
+    return environment_id_options.get(environment_id)
+
+
+class Client(BaseClient):
+    """
+    Client to use in the CrowdStrikeFalconX integration. Uses BaseClient
+    """
+
+    def __init__(self, server_url: str, username: str, password: str, use_ssl: bool, proxy: bool):
+        super().__init__(base_url=server_url, verify=use_ssl, proxy=proxy)
+        self._username = username
+        self._password = password
+        self.token = self._generate_token()
+        self._headers = {'Authorization': 'bearer ' + self.token}
+
+    def _generate_token(self) -> str:
+        """Generate an Access token using the user name and password
+        :return: valid token
+        """
+        body = {
+            'client_id': self._username,
+            'client_secret': self._password
+        }
+
+        byte_creds = f'{self._username}:{self._password}'.encode('utf-8')
+
+        headers = {
+            'Authorization': f'Basic {base64.b64encode(byte_creds).decode()}'
+        }
+        token_res = self._http_request('POST', '/oauth2/token', data=body, headers=headers)
+        return token_res.get('access_token')
+
+    def upload_file(
+            self,
+            file: str,
+            file_name: str,
+            is_confidential: str = "true",
+            comment: str = ""
+    ) -> Union[dict, str, Response]:
+        """Creating the needed arguments for the http request
+        :param file: content of the uploaded sample in binary format
+        :param file_name: name of the file
+        :param is_confidential: defines visibility of this file in Falcon MalQuery, either via the API or the Falcon console
+        :param comment: a descriptive comment to identify the file for other users
+        :return: http response
+        """
+        shutil.copy(demisto.getFilePath(file)['path'],
+                    demisto.getFilePath(file)['name'])
+        f = open(demisto.getFilePath(file)['name'], 'rb')
+        url_suffix = f"/samples/entities/samples/v2?file_name={file_name}&is_confidential={is_confidential}&comment={comment}"
+        self._headers['Content-Type'] = 'application/octet-stream'
```

Comment:
before the return, use shuitil to rm the file.
e.g:
    finally:
        shutil.rmtree(file_name, ignore_errors=True)